tests/components/backup_sftp/test_config_flow.py

```diff
@@ -0,0 +1,247 @@
+"""Tests config_flow."""
+
+from collections.abc import Awaitable, Callable
+from tempfile import NamedTemporaryFile
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from asyncssh.misc import PermissionDenied
+from asyncssh.sftp import SFTPNoSuchFile, SFTPPermissionDenied
+import pytest
+
+from homeassistant.components.backup_sftp.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PRIVATE_KEY_FILE,
+    CONF_USERNAME,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.exceptions import ConfigEntryError
+
+from .conftest import USER_INPUT
+
+from tests.common import MockConfigEntry
+
+type ComponentSetup = Callable[[], Awaitable[None]]
+
+
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_backup_sftp_full_flow(
+    hass: HomeAssistant,
+    fake_connect: AsyncMock,
+) -> None:
+    """Test the full backup_sftp config flow with valid user input."""
+    # Start the configuration flow
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    # The first step should be the "user" form.
+    assert result["step_id"] == "user"
+
+    # Patch the BackupAgentClient so that when the flow creates it, our fake is used.
+    with (
+        patch(
+            "homeassistant.components.backup_sftp.config_flow.connect",
+            return_value=fake_connect,
+        ) as mock_client,
+        patch(
+            "homeassistant.components.backup_sftp.config_flow.SSHClientConnectionOptions",
+            return_value=MagicMock(),
+        ),
+        NamedTemporaryFile() as tmpfile,
+    ):
+        # Create a tempfile so we don't get `ConfigEntryError` during setup.
+        user_input = USER_INPUT.copy()
+        user_input[CONF_PRIVATE_KEY_FILE] = tmpfile.name
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input
+        )
+        await hass.async_block_till_done()
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(mock_client.mock_calls) == 1
+
+    # Verify that a new config entry is created.
+    assert result["type"] == FlowResultType.CREATE_ENTRY
```

Comment:
```suggestion
    assert result["type"] is FlowResultType.CREATE_ENTRY
```
`is` for enum comparison (everywhere else as well)

Reply:
this one is still open

Reply:
Sorry, missed this one. Done.